### DISCUSSION TITLE
Trivy does not detect CVE-2023-3635 against okio-jvm-3.0.0.jar

### DISCUSSION THREAD

**Discussion by navzen2000** [aquasecurity/trivy#5035] [Q&A]
### Description

Trivy does not detect CVE-2023-3635 against okio-jvm-3.0.0.jar
Other tools like Sonatype is detecting it

### Desired Behavior

CVE-2023-3635 to be reported against okio-jvm-3.0.0.jar

### Actual Behavior

Trivy did not report

### Reproduction Steps

```bash
Run Trivy against a container with  okio-jvm-3.0.0.jar
```

### Target

Container Image

### Scanner

Vulnerability

### Output Format

None

### Mode

None

### Debug Output

```bash
Not available
```

### Operating System

linux

### Version

```bash
Version: 0.44.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-08-24 06:08:49.320860551 +0000 UTC
  NextUpdate: 2023-08-24 12:08:49.320859751 +0000 UTC
  DownloadedAt: 2023-08-24 07:13:38.403607124 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-08-24 00:51:26.692869902 +0000 UTC
  NextUpdate: 2023-08-27 00:51:26.692869302 +0000 UTC
  DownloadedAt: 2023-08-24 04:57:05.068649297 +0000 UTC
```

### Checklist

- [ ] Run `trivy image --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by nikpivkin** [comment]
Hi @navzen2000 !

The package `com.squareup.okio:okio` is specified in [CVE-2023-3635](https://github.com/advisories/GHSA-w33c-445m-f8w7), so Trivy does not detect this vulnerability for `com.squareup.okio:okio-jvm`. I think you need to create an issue or PR in the [GitHub Advisory Database](https://github.com/github/advisory-database), since okio is a multiplatform project and has builds for many platforms: jvm, js, etc. and maven defines them as different artifacts.

**Reply by navzen2000** [reply]
Please refer to matching CPE in [NVD](https://nvd.nist.gov/vuln/detail/CVE-2023-3635#range-9412322)
[cpe:2.3:a:squareup:okio:3.0.0:-:*:*:*:*:*:*](https://nvd.nist.gov/products/cpe/detail/1436860?keyword=cpe%3A2.3%3Aa%3Asquareup%3Aokio%3A3.0.0%3A-%3A*%3A*%3A*%3A*%3A*%3A*&status=FINAL,DEPRECATED&orderBy=CPEURI&namingFormat=2.3)

Also, Sonatype is able to recognize it.

**Reply by nikpivkin** [reply]
CPE refers to the artifact [com.squareup.okio:okio](https://mvnrepository.com/artifact/com.squareup.okio/okio), but `okio-jvm-3.0.0.jar ` is the artifact file [com.squareup.okio:okio-jvm](https://mvnrepository.com/artifact/com.squareup.okio/okio-jvm). Sonatype [shows](https://ossindex.sonatype.org/component/pkg:maven/com.squareup.okio/okio-jvm@3.0.0) that there are no vulnerabilities in `okio-jvm`.

**Comment by navzen2000** [comment]
As per Github commit of the fix, it went into okio/src/jvmMain/kotlin/okio/GzipSource.kt
https://github.com/square/okio/commit/81bce1a30af244550b0324597720e4799281da7b

com.squareup.okio:okio-jvm contains GzipSource.class

java -jar jd-cli-1.2.1.jar GzipSource.class
21:03:28.003 INFO com.github.kwart.jd.cli.Main - Decompiling GzipSource.class
import java.io.EOFException;
import java.io.IOException;
import java.util.Arrays;
import java.util.zip.CRC32;
import java.util.zip.Inflater;
import kotlin.Metadata;
import kotlin.jvm.internal.Intrinsics;
import okio.Buffer;
import okio.BufferedSource;
import okio.GzipSource;
import okio.InflaterSource;
import okio.RealBufferedSource;
import okio.Segment;
import okio.Source;
import okio.Timeout;
import org.jetbrains.annotations.NotNull;
